user-interface/src/lib/models/api2.ts

```diff
@@ -198,19 +200,21 @@ type CacheOptions = {
 };
 
 function withCache(cacheOptions: CacheOptions): Pick<GenericApiClient, 'get'> {
+  const key = API_CACHE_NAMESPACE + cacheOptions.key;
+
   // TODO: For now we are only implementing `get`. In the future we may want to cache responses from the other HTTP verbs.
```

Comment:
I left this note a while ago. I don't think we should cache any verb but GET anymore.
```suggestion
  // TODO: For now we are only implementing `get`. In the future we may want to cache responses from the other HTTP verbs.
```